src/main/java/com/github/manolo8/darkbot/gui/players/PlayerEditor.java

```diff
@@ -4,29 +4,71 @@
+import eu.darkbot.api.events.Listener;
+import eu.darkbot.api.game.entities.Player;
+import eu.darkbot.api.managers.EntitiesAPI;
+import eu.darkbot.api.managers.EventBrokerAPI;
 import net.miginfocom.swing.MigLayout;
 
 import javax.swing.*;
-import java.awt.event.ActionEvent;
-import java.util.Comparator;
-import java.util.List;
-import java.util.Locale;
+import java.util.*;
 
-public class PlayerEditor extends JPanel {
+import static java.util.stream.Collectors.toList;
+
+public class PlayerEditor extends JPanel implements Listener {
     private final JList<PlayerInfo> playerInfoList;
     private final DefaultListModel<PlayerInfo> playersModel;
+    private final List<PlayerInfo> nearbyPlayerList = new ArrayList<>();
 
     private final PlayerManager playerManager;
     private final SearchField sf;
     private final PlayerTagManager tagEditor;
     protected Main main;
 
+    @EventHandler
+    public void onEntityPlayerAdd(EntitiesAPI.EntityCreateEvent event) {
+        if (event.getEntity() instanceof Player) {
+            if (playersModel != null && main.config.PLAYER_INFOS != null && main.config.PLAYER_INFOS.values()
+                    .stream().noneMatch(a -> a.toString()
+                            .contains(((Player) event.getEntity()).getEntityInfo().getUsername()))) {
+                nearbyPlayerList.add(new PlayerInfo(((Player) event.getEntity())
+                        .getEntityInfo().getUsername(), event.getEntity().getId()));
+
+                SwingUtilities.invokeLater(() -> playersModel.
+                        addElement(new PlayerInfo(((Player) event.getEntity())
+                                .getEntityInfo().getUsername(), event.getEntity().getId())));
+                System.out.println("added " + ((Player) event.getEntity()).getEntityInfo().getUsername());
+            } else {
+                System.out.println("in list yet " + ((Player) event.getEntity()).getEntityInfo().getUsername());
+            }
+        }
+    }
+
+    @EventHandler
+    public void onEntityPlayerRemove(EntitiesAPI.EntityRemoveEvent event) {
+        if (event.getEntity() instanceof Player) {
+            if (playersModel != null && main.config.PLAYER_INFOS != null && main.config.PLAYER_INFOS.values()
+                    .stream()
+                    .noneMatch(a -> a.toString()
+                            .contains(((Player) event.getEntity()).getEntityInfo().getUsername()))) {
+                nearbyPlayerList.stream()
+                        .filter(a -> a.toString().contains(((Player) event.getEntity()).getEntityInfo().getUsername()))
+                        .collect(toList()).forEach(nearbyPlayerList::remove);
+                
+                SwingUtilities.invokeLater(() -> Arrays.stream(playersModel.toArray())
+                        .filter(a -> a.toString().contains(((Player) event.getEntity()).getEntityInfo().getUsername()))
+                        .forEach(playersModel::removeElement));
```

Comment:
Same issues as above, but you're also allocating a new array for the whole model, including both nearby and non-nearby players.

Iterate the model to find the object,, and call removeElement for just that, also, update the nearby player list in the swing thread to avoid race conditions:

```suggestion
                SwingUtilities.invokeLater(() -> {
                        for (int i = 0; i < playersModel.size(); i++) {
                            PlayerInfo info = playersModel.getElement(i);
                            if (info.getId() == event.getEntity().getId()) {
                                nearbyPlayerList.remove(info);
                                playersModel.removeElement(info);
                                break;
                            }
                        }
                });
```
Keep in mind this will only work if the player info you use in both lists is the same, which you SHOULD do (see the comment on player add)